### PR TITLE
Tasks 81 90

### PR DESCRIPTION
**Pull request by dudzon**

### CODE REVIEW

File: tasks/82_bouncing-balls/bouncing-balls.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const bouncingBall = ( h, bounce, window ) => {
+  if ( h <= 0 || bounce <= 0 || bounce >= 1 || window >= h ) return -1;
+  let result = 1;
+  while (( h *= bounce ) > window ) {
+    result += 2;
+  }
```
Comment (tkachukivan) [inline tasks/82_bouncing-balls/bouncing-balls.js]: I've also solve this as you, but there is a way to solve this without loop:)
`Math.floor(Math.log(w/h)/Math.log(b)) * 2 + 1`

File: tasks/83_missing-term/missing-term.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable max-len */
+// I don't know if the way I calculated step in diffs array is correct,
```
Comment (tkachukivan) [inline tasks/83_missing-term/missing-term.js:2]: Generally you also can solve it by loop from 1 element and if `arr[i] - arr[i - 1] !== arr[i + 1] - arr[i]` than missed element will equal `arr[i] + smallestDiff`

File: tasks/89_symmertic-difference/symmertic-difference.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      result.push( element );
+    } else {
+      b.splice( b.indexOf(element), 1 );
+    }
+  });
+  return result.concat(b);
+};
+
+export const sym = (...args) => {
+  let symmetricDifference = symDiff( args[0], args[1] );
```
Comment (tkachukivan) [inline tasks/89_symmertic-difference/symmertic-difference.js]: why not use reduce method instead?

File: tasks/89_symmertic-difference/symmertic-difference.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable max-len */
+const symDiff = (a, b) => {
+  const result = [];
+
+  a = [...new Set( a.sort() )];
+
+  b = [...new Set( b.sort() )];
+
+  a.map(( element ) => {
+    if ( b.indexOf( element ) === -1 ) {
```
Comment (tkachukivan) [inline tasks/89_symmertic-difference/symmertic-difference.js]: result if indexOf better save to variable

**Review by tkachukivan — APPROVED** [review]

File: tasks/86_working-hours/working-hours.js [hunk 5]
```diff
@@ -1,30 +1,71 @@
...
-      if ( workday && !first ) {
-        first = workday.day;
-        currentFrom = workday.from;
-        currentTo = workday.to;
-      } else if (first) {
-        last = day;
+    for ( let i = 0; i < days.length; i++) {
+      let workday = result.find( (element) => element[0] === days[i]);
+      switch ( workday[1]) {
+        case '11:00':
```
Comment (tkachukivan) [inline tasks/86_working-hours/working-hours.js]: from my point of view it is not the best solution, that if workday will start at 9? at 8? at 7? it is not very good thing to relay to some values as they can change easily

**Review by tkachukivan — APPROVED** [review]

File: tasks/82_bouncing-balls/bouncing-balls.js (hunk 1 above)
Comment (dudzon) [reply tasks/82_bouncing-balls/bouncing-balls.js]: done

File: tasks/89_symmertic-difference/symmertic-difference.js (hunk 4 above)
Comment (dudzon) [reply tasks/89_symmertic-difference/symmertic-difference.js]: Done

File: tasks/86_working-hours/working-hours.js (hunk 5 above)
Comment (dudzon) [reply tasks/86_working-hours/working-hours.js]: Agreed :)

File: tasks/89_symmertic-difference/symmertic-difference.js (hunk 3 above)
Comment (dudzon) [reply tasks/89_symmertic-difference/symmertic-difference.js]: Done :)

File: tasks/86_working-hours/working-hours.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+//       } else if (first) {
+//         last = day;
+//       }
+//     }
+//     return result;
+//   }
+//   return [];
+// };
+
+//  Solution 2: It is really shitty code, with harcoded values but it's with grouping as Misha wanted me to do.
```
Comment (mikhail-ev) [inline tasks/86_working-hours/working-hours.js:32]: Seems like misunderstanding I thought we agreed to group like "if the current day starts/ends at the same time then it's a group". So please get rid of case statement

**Review by mikhail-ev — CHANGES_REQUESTED** [review]

File: tasks/86_working-hours/working-hours.js (hunk 6 above)
Comment (dudzon) [reply tasks/86_working-hours/working-hours.js:32]: Done

File: tasks/88_record-collection/record-collection.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  '5439': {
+    'album': 'ABBA Gold',
+  },
+};
+
+export const updateRecords = ( id, prop, value ) =>{
+  if ( collection.hasOwnProperty(id)) {
+    if ( prop !== 'tracks' && value !== '') {
+      collection[id][prop] = value;
+    } else if ( prop === 'tracks' && !collection[id][prop] ) {
```
Comment (tsvetkovv) [inline tasks/88_record-collection/record-collection.js]: You can simplify by union of `prop === 'tracks'` condition

Comment (dudzon) [reply tasks/88_record-collection/record-collection.js]: Done

**Pull request merged**